packages/components/organisms/f-footer/test-utils/component-objects/f-footer.component.js

```diff
@@ -1,36 +1,84 @@
-import {
+const Page = require('@justeat/f-wdio-utils/src/page.object');
+const {
     FOOTER_COMPONENT,
     FOOTER_ICONS,
-    COURIER_LINKS
-} from './f-footer.selectors';
-
-// Component
-const footerComponent = () => $(FOOTER_COMPONENT);
-const footerIcons = () => $$(FOOTER_ICONS);
-const courierLinks = () => $(COURIER_LINKS);
-
-// Icons
-const iosIcon = () => footerIcons()[0];
-const androidIcon = () => footerIcons()[1];
-const huaweiIcon = () => footerIcons()[1];
-const facebookIcon = () => footerIcons()[3];
-const twitterIcon = () => footerIcons()[4];
-const youtubeIcon = () => footerIcons()[5];
-
-// Functions
-exports.isFooterDisplayed = () => footerComponent().isDisplayed();
-exports.isIosIconDisplayed = () => iosIcon().isDisplayed();
-exports.isAndroidIconDisplayed = () => androidIcon().isDisplayed();
-exports.isHuaweiIconDisplayed = () => huaweiIcon().isDisplayed();
-exports.isFacebookIconDisplayed = () => facebookIcon().isDisplayed();
-exports.isTwitterIconDisplayed = () => twitterIcon().isDisplayed();
-exports.isYoutubeIconDisplayed = () => youtubeIcon().isDisplayed();
-exports.waitForFooter = () => footerComponent().waitForExist();
-exports.isCourierLinksDisplayed = () => courierLinks().isDisplayed();
-
-// Click Function
-exports.clickIosIcon = () => iosIcon().click();
-exports.clickAndroidIcon = () => androidIcon().click();
-exports.clickFacebookIcon = () => facebookIcon().click();
-exports.clickTwitterIcon = () => twitterIcon().click();
-exports.clickYoutubeIcon = () => youtubeIcon().click();
+    DOWNLOAD_ICONS,
+    COURIER_LINKS,
+    SOCIAL_ICONS
+} = require('./f-footer.selectors');
+
+class Footer extends Page {
+
+    get component () { return $(FOOTER_COMPONENT) }
+    get icons () { return  $$(FOOTER_ICONS) }
+    get downloadIcons () { return $$(DOWNLOAD_ICONS) }
+    get socialIcons () { return $$(SOCIAL_ICONS) }
+    get courierLinks () { return $(COURIER_LINKS) }
+    get downloadIcon ()  { return this.downloadIconValue }
+    get socialIcon () { return this.socialIconValue }
+
+    set expectedDownloadIcon(icon) {
+        this.downloadIconValue = this.downloadIcons.filter(element => element.getAttribute('data-test-id').includes(icon))[0];
+    }
+
+    set expectedSocialIcon(icon){
+        this.socialIconValue = this.socialIcons.filter(element => element.getAttribute('data-test-id').includes(icon))[0];
+    }
+
+    open(){
```

Comment:
I'd delete this function, and rename `openWithLocale` to `open` and have a default parameter

like so:


```
    openWithLocale(locale = 'gb'){
        let countryFormatted = locale.toUpperCase();
        let formattedLocale = '';
        switch ( countryFormatted ){
            case 'GB':
            case 'AU':
                formattedLocale = `en-${countryFormatted}`
                break
            default: 
                throw new Error (`locale ${countryFormatted} is not supported`);
        }
        super.openComponent('organism', `footer-component&knob-Locale=${formattedLocale}`);
    }
```